src/lib/utils.ts

```diff
@@ -37,29 +39,124 @@ export function promisifyServiceClient(client: Client) {
   });
 }
 
-interface FieldMask {
-  pathsList: string[];
+function isValueField(isObject: boolean, value: any): boolean {
+  if (typeof value === "undefined") {
+    return false;
+  }
+  return isObject ? value.hasOwnProperty("value") && Object.keys(value).length === 1 : false;
 }
 
-export function formatCallResults(resultsList: any[], fieldMask: FieldMask | undefined) {
-  const parsedResults: any[] = [];
+function getNestedValue(path: string, data: GoogleAdsRow): any {
+  const value = get(data, path) ?? get(data, `${path}List`);
+  if (typeof value === "undefined") {
```

Comment:
This function (getNestedValue) is called lots and lots, so efforts here to make it run faster (at the expense of legibility) could be worthwhile. A few ideas, in case you haven't benchmarked these already:
- instead of calling `typeof value` several times, store that string in a variable for comparisons.
- inline `isValueField`, which would allow for removing the inline if-statement on `isObject`
- Consider removing `if (value.hasOwnProperty(childKey)) {`, which I don't think will ever be false